src/test/java/guitests/AddCommandTest.java

```diff
@@ -75,42 +104,59 @@ public void add_unknownCommandName_errorMessage() {
     }
 
     @Test
-    public void add_invalidName_errorMessage() {
+    public void add_tabAutocompleteFromOneChar_success() {
+        assertAddTabAutocomplete(AddCommand.COMMAND_WORD.substring(0, 1));
+    }
+
+    @Test
+    public void add_tabAutocompleteFromTwoChars_success() {
+        assertAddTabAutocomplete(AddCommand.COMMAND_WORD.substring(0, 2));
+    }
+
+    @Test
+    public void add_invalidCommandFormat_errorMessage() {
         commandBox.runCommand("add");
         assertResultMessage(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
 
         commandBox.runCommand("add ");
         assertResultMessage(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
 
-        commandBox.runCommand("add /");
-        assertResultMessage(Name.MESSAGE_NAME_CONSTRAINTS);
+        commandBox.runCommand("add /tag");
+        assertResultMessage(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
+    }
 
+    @Test
+    public void add_invalidEntryName_errorMessage() {
         commandBox.runCommand("add $");
         assertResultMessage(Name.MESSAGE_NAME_CONSTRAINTS);
 
-        commandBox.runCommand("add ?");
-        assertResultMessage(Name.MESSAGE_NAME_CONSTRAINTS);
-
         commandBox.runCommand("add /ta");
         assertResultMessage(Name.MESSAGE_NAME_CONSTRAINTS);
-
-        commandBox.runCommand("add /tag");
-        assertResultMessage(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
     }
 
     @Test
     public void add_invalidTags_errorMessage() {
         commandBox.runCommand("add task /tag");
         assertResultMessage(Tag.MESSAGE_TAG_CONSTRAINTS);
 
-        commandBox.runCommand("add task /tag /");
-        assertResultMessage(Tag.MESSAGE_TAG_CONSTRAINTS);
-
         commandBox.runCommand("add task /tag $");
         assertResultMessage(Tag.MESSAGE_TAG_CONSTRAINTS);
+    }
 
-        commandBox.runCommand("add task /tag ?");
-        assertResultMessage(Tag.MESSAGE_TAG_CONSTRAINTS);
+    private void assertAddTabAutocomplete(String input) {
```

Comment:
Maybe add a header comment here? It's not very clear from the method name what this method is doing